test/block.test.js

```diff
@@ -39,12 +40,70 @@ statusSuccess.title = (providedTitle = '', input) =>  `Successfully calls ${inpu
+
+const returnsNull = async (t, { method, params = {} }) => {
+    const response = await t.context.web3data.block[method](999999999)
+    t.is(response, null)
+}
+returnsNull.title = (providedTitle = '', input) => `Successfully calls ${input.method} and returns null value`
+
+const returnsBlockObject = async (t, { method, params = {} }) => {
+    const block = await t.context.web3data.block[method](params.id)
+    t.is(parseInt(block.number), params.id)
+    t.is(typeof block, 'object')
+}
+returnsBlockObject.title = (providedTitle = '', input) => `Successfully calls ${input.method} and returns valid block object`
+
+const returnsBlockObjectFilters = async (t, { method, params = {} }) => {
+    const {id, timeFormat, validationMethod} = params
+    const block = await t.context.web3data.block[method](id, {timeFormat, validationMethod})
+    t.is(parseInt(block.number), params.id)
+    t.true(_.has(block, 'validation'))
+    t.is(typeof block, 'object')
+}
+returnsBlockObjectFilters.title = (providedTitle = '', input) => `Successfully calls ${input.method} with filters returns valid block object`
+
+const returnsUncleObject = async (t, { method, params = {} }) => {
+    const uncle = await t.context.web3data.block[method](params.id, params.index)
+    t.is(parseInt(uncle.blockNumber), params.id)
+    t.is(typeof uncle, 'object')
+}
+returnsUncleObject.title = (providedTitle = '', input) => `Successfully calls ${input.method} and returns valid uncle object`
+
+const returnsTxnObjects = async (t, { method, params = {}}) => {
+    const transactions = await t.context.web3data.block[method](params.id)
+    t.true(transactions.length > 0)
+    t.true(_.has(transactions[0], 'blockNumber'))
+    t.is(parseInt(transactions[0].blockNumber), params.id)
+}
+returnsTxnObjects.title = (providedTitle = '', input) => `Successfully calls ${input.method} and returns array of valid txn objects`
+
+const returnsTxnObject = async (t, { method, params = {} }) => {
+    const transaction = await t.context.web3data.block[method](params.id, params.index)
+    t.true(_.has(transaction, 'hash'))
+    t.is(parseInt(transaction.blockNumber), params.id)
+    t.is(parseInt(transaction.index), params.index)
+}
+returnsTxnObject.title = (providedTitle = '', input) => `Successfully calls ${input.method} and returns valid txn object`
+
+// test([statusSuccess, rejectsPromise],  {method: 'getTokenTransfers'}, NO_NUMBER)
+test([returnsBlockObject, returnsBlockObjectFilters], {method: 'getBlock', params: {id: 7000000, timeFormat: 'ms', validationMethod: 'full'}})
+test([returnsNumber], {method: 'getBlockNumber'})
+test([returnsNumber, returnsNull, rejectsPromise], {method: 'getBlockTransactionCount', params: {id: 7000000}}, NO_BLOCK_ID)
+test([returnsUncleObject, returnsNull, rejectsPromise], {method: 'getUncle', params: {id: 8102326, index: 0}}, NO_BLOCK_ID)
+test([returnsTxnObjects], {method: 'getTransactions', params: {id: 8102326}}, NO_BLOCK_ID)
```

Comment:
@taylorjdawson again, let's keep these simple, try not to refactor tests too much please